src/google/protobuf/generated_message_tctable_impl.h

```diff
@@ -269,7 +269,7 @@ enum FieldType : uint16_t {
 
 #ifndef NDEBUG
 template <size_t align>
-void AlignFail(uintptr_t address) {
+[[noreturn]] void AlignFail(uintptr_t address) {
```

Comment:
I don't think all the compilers in our toolchain support a raw [[noreturn]] atribute.  I think you need to do something like

https://github.com/abseil/abseil-cpp/blob/master/absl/base/attributes.h#L195-L204

in port_def.inc and port_undef.inc

Reply:
Well it was used for all but msvc before it was removed, so that tells me that every compiler in your suite supported it, except maybe an ancient MSVC version?

Reply:
I would feel more comfortable using the same attributes as we do for abseil.  Protobuf is in the (slow) process of rationalizing all its supported platforms and standardizing on the same ones as protobuf and grpc.

Reply:
I don't seem to be able to find any information on when Microsoft started to support `[[noreturn]]` in the standard form. It has been in C++11 which means it is likely supported since VS 2017. However I don't have that available to test and the earliest one in compiler explorer is 19.14, which supports it.
Since MSVC doesn't complain about it anyway (and does not seem to generate any different code on older versions), we could just go ahead and only have the attribute on newer MSVC versions.

Reply:
I don't recall any bugs filed against absl.  Feels like we can use there version with risk (the licenses are compatible)